docs/architecture/adr-100-data-companion-push-api.md

```diff
@@ -0,0 +1,441 @@
+
+# Use the experimental gzip compression call option when submitting data to the
+# server. See https://pkg.go.dev/google.golang.org/grpc#UseCompressor
+experimental_use_gzip = false
+
+# Controls the BlockCommitted gRPC call.
+[data_companion.block_committed]
+# Enable the BlockCommitted gRPC call. Only relevant if the data companion is
+# enabled.
+enabled = true
+# Additional fields to publish in each BlockCommittedRequest sent to the
+# companion. Available options:
+# - "finalize_block_response": Also publish the FinalizeBlock response related
+#                              to the block in the BlockCommittedRequest.
+additionally_publish = ["finalize_block_response"]
+```
+
+It is unclear at present whether compressing the data being sent to the
+companion service will result in meaningful benefits.
+
+### Monitoring
+
+To monitor the health of the interaction between Tendermint and the companion
+service, the following additional Prometheus metrics are proposed:
+
+- `data_companion_send_time` - A gauge that indicates the maximum time, in
+  milliseconds, taken to send a single request to the companion service take
+  from a rolling window of tracked send times (e.g. the maximum send time over
+  the past minute).
+- `data_companion_send_bytes` - A counter indicating the number of bytes sent to
+  the data companion since node startup.
+- `data_companion_buffer_size` - A gauge indicating the number of requests
+  currently being buffered on disk on behalf of the companion service for each
+  kind of request.
+- `data_companion_buffer_size_bytes` - A gauge indicating how much data is being
+  buffered by Tendermint on disk, in bytes, on behalf of the companion service,
+  for each kind of request.
+
+## Implications
+
+1. We will be able to mark the following RPC APIs as deprecated and schedule
+   them for removal in a future release:
+   1. The [WebSocket subscription API][websocket-api]
+   2. [`/tx_search`]
+   3. [`/block_search`]
+   4. [`/block_results`]
+
+   We may eventually be able to remove the [`/broadcast_tx_commit`] endpoint,
+   but this is often useful during debugging and development, so we may want to
+   keep it around.
+
+2. We will be able to remove all event indexing from Tendermint once we remove
+   the above APIs.
+
+3. Depending on the implementation approach chosen, we will still need to store
+   some quantity of data not critical to consensus. This data can automatically
+   be pruned once Tendermint has successfully transmitted it to the companion
+   service.
+
+### Release Strategy
+
+As captured in the [requirements](#requirements), we should be able to release
+this as an additive, opt-in change, thereby not impacting existing APIs. This
+way we can evolve this API until such time that consumers are satisfied with
+removal of the old APIs mentioned in [implications](#implications).
+
+This also means we could potentially release this as a non-breaking change (i.e.
+in a patch release).
```

Comment:
If nobody (i.e., no user) is asking for this, I wouldn't promise anything along these lines.
It will already be delicate enough to release it in a smooth way in a new major release.